packages/reporting/src/sanitizer.js

```diff
@@ -314,8 +314,18 @@ const RISKY_URL_PATH_PARTS = new Set([
  *     perfect, but quorum should be used as an additional step; quorum is effective
  *     in dropping unique identifiers like unique tokens that may slip through
  *     static rules.
+ * - tryPreservePath [default=false]:
+ *     If the URL fails and gets truncated, it will normally keep only keep the
+ *     the scheme and the domain. With this option, it does an extra pass to
+ *     test if adding the URL path would be safe.
+ *     For instance, given "https://example.test/foo/bar?arg=1&...#hash..."),
+ *     it will check again with "https://example.test/foo/bar". If the latter URL
+ *     passes all (strict) checks, it will be used for the truncated URL; instead of
+ *     "https://example.test (PROTECTED)", the result will be
+ *     "https://example.test/foo/bar (PROTECTED)".
  */
-export function sanitizeUrl(url, { strict = false } = {}) {
+export function sanitizeUrl(url, options = {}) {
+  const { strict = false, tryPreservePath = false } = options;
```

Comment:
`export function sanitizeUrl(url, { strict = false, tryPreservePath = false } = {}) {`

Reply:
The options are captured to be later forwarded in the recursive call.